kgraphql-ktor/src/main/kotlin/com/apurebase/kgraphql/KtorFeature.kt

```diff
@@ -87,12 +87,17 @@ class GraphQL(val schema: Schema) {
                             )
                             call.respondText(result, contentType = ContentType.Application.Json)
                         }
-                        if (config.playground) get {
-                            @Suppress("RECEIVER_NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
-                            val playgroundHtml =
-                                KtorGraphQLConfiguration::class.java.classLoader.getResource("playground.html")
-                                    .readBytes()
-                            call.respondBytes(playgroundHtml, contentType = ContentType.Text.Html)
+                        get {
+                            val schemaRequested = call.request.queryParameters["schema"] != null
+                            if (schemaRequested && config.introspection) {
+                                call.respondText(schema.printSchema())
+                            } else if (config.playground) {
+                                @Suppress("RECEIVER_NULLABILITY_MISMATCH_BASED_ON_JAVA_ANNOTATIONS")
+                                val playgroundHtml =
+                                    KtorGraphQLConfiguration::class.java.classLoader.getResource("playground.html")
```

Comment:
Off topic:

Is this reading the resource on every invocation? Or is there an internal caching mechanism?